src/metabase/api/user.clj

```diff
@@ -84,7 +84,7 @@
   {password     [Required ComplexPassword]}
   (check-self-or-superuser id)
   (let-404 [user (sel :one [User :password_salt :password] :id id :is_active true)]
-    (when-not (:is_superuser @*current-user*)
+    (when (= id (:id @*current-user*))
```

Comment:
Not sure I understand what was/is going on here. Previously we were only requiring the old password if it was a superuser changing the password (which doesn't make sense), now we require the password if any user is changing their own password?